database-plugins/src/main/java/co/cask/hydrator/plugin/DBUtils.java

```diff
@@ -99,6 +100,29 @@ public static DriverCleanup ensureJDBCDriverIsAvailable(Class<? extends Driver>
    * list of {@link co.cask.cdap.api.data.schema.Schema.Field},
    * where name of the field is same as column name and type of the field is obtained using {@link DBUtils#getType(int)}
    * @param resultSet result set of executed query
+   * @param schemaStr schema string to override resultant schema
+   *
+   * @return list of schema fields
+   * @throws SQLException
+   * @throws IOException
+   */
+  public static List<Schema.Field> getSchemaFields(ResultSet resultSet, String schemaStr) throws SQLException {
+    if (!Strings.isNullOrEmpty(schemaStr)) {
+      try {
+        Schema schema = Schema.parseJson(schemaStr);
+        return schema.getFields();
```

Comment:
we should still get the fields from the result set. We should then check that the user provided schema is compatible with the result schema. Each field in the user provided schema exists in the result schema and is the same type, or compatible non-nullable type.